apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.py

```diff
@@ -39,8 +39,10 @@ def ansi_to_html(output: str | None) -> str | None:
         return None
     try:
         return conv.convert(output, full=False)
-    except Exception as exc:
+    except (ValueError, RuntimeError) as exc:
         return f"[Error converting ANSI to HTML: {exc}]\n\n{output}"
+    except Exception as exc:  # noqa: BLE001
+        return f"[Unknown error converting ANSI to HTML: {exc}]\n\n{output}"
```

Comment:
I truly don't see the value here. What difference could it possibly make to a user whether the error is "unknown" or not? Ditto elsewhere in the PR.

Reply:
If an error says "Unknown error", as a user this flags to me that this is a bug in the software. If an error says "error", it just flags that I did something wrong.

Reply:
In my (dare I say PL's) experience, people see the word "error" and, depending on their personality, always immediately blame either themselves or the system 😅

Reply:
I'm still deciding how I feel about this. At a minimum, I lean towards not being dogmatic about this and disabling the rule in cases like this where, IMO, neither we nor users derive value from the extra code paths involved. As probably our main Python expert, I'm curious to hear what @eliotwrobson thinks.

Reply:
Agree with Nate, there's not really a lot of value in the additional specificity here. A lint rule like this is wayyyy more useful in, for example, large application code where different exceptions can be responded to with different types of logging messeges. For our use case, people don't even care that much about the specific exception type (which is why we've been getting away with having a generic exception in a lot of places until this PR was introduced).

Reply:
Reverted some of the more egregious cases in 55f351fea4dc4304cdede5f9a3f902b484f6ebf7. Some guards like

```
try:
   x = int(y)
except ValueError:
   ...
``` 
I kept.